### ISSUE TITLE
Improvement on pylint score to 8.16

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#894]
_From [fhe...@gmail.com](https://code.google.com/u/111731474599148743921/) on 2011-12-22T12:15:21Z_

What steps will reproduce the problem?

 1.execute pylint bootstrap.py 2. 3. What is the expected output? What do you see instead? The core rate for the code should be at least 8 out of 10. But the output reports 4.9 out of 10.

with this patch, the reported score is at 8.16 out of 10. What version of the product are you using? On what operating system? current mercurial version of spyderlib. Macos X 10.7.2 

Please provide any additional information below

.

**Attachment:** [bootstrap.py_pylintscore8.16.patch](http://code.google.com/p/spyderlib/issues/detail?id=894)

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=894_

**Comment by spyder-issues-migrator** [comment]
_From [techtonik@gmail.com](https://code.google.com/u/techtonik@gmail.com/) on 2011-12-27T03:24:26Z_

I may take a look at this as soon as I have time to install and play with `pylint`. I wonder why all arguments are placed in __doubleunderlines__?

**Status:** Accepted  
  **Owner:** techtonik@gmail.com

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2011-12-27T06:10:03Z_

According to Python development guidelines, globals should not be lowercased. That's why for example `options` has to be replaced by:
  * `OPTION` (I prefer this: less characters means less effort when writing code and less typo errors...)
  * or `__option__`.

But, this is a *guideline*: when writing scripts (like bootstrap.py or clonebuild.py), even the most experienced Python programmers won't respect this rule and will write global references with lower-cased and non-double-underscore names. This kind of rule is made to be broken in such circumstances where breaking them do not prevent the script from being easy to read, to debug and to maintain.

In other words, this very rule was originally made for library-coding (i.e. Python modules within libraries, not individual scripts like bootstrap.py) because its application for individual scripts would be to much a constraint and would make the script less easy to read (which is the opposite of what these coding guidelines were made for).

So, I suggest using the following pylint exception rule:
\# pylint: disable=C0103
(like in spyderlib/widgets/explorer.py)

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2011-12-27T06:15:25Z_

This issue was closed by revision b9ff1a4bc95f .

**Status:** Fixed
